code/src/types/product.types.ts

```diff
@@ -54,6 +56,23 @@ export type GetUserProductByIdParams = Lang & {
 export type GetManagerProductsResponse = PageableResponse<ManagerProduct[]>;
 
 export type GetManagerProductsParams = Partial<Pageable & Lang>;
+export type GetUserProductsBySearchQueryParams = Lang & {
+  page?: number;
+  size?: number;
+  sort?: string;
+  searchQuery: string;
+};
+export type GetUserProductsBySearchQueryResponse = {
+  content: ProductFromSearch[];
+  totalElements: number;
+  totalPages: number;
+  first: boolean;
+  last: boolean;
+  number: number;
+  numberOfElements: number;
+  size: number;
+  empty: boolean;
+};
```

Comment:
I created reusable type for this case:

```tsx
export type PageableResponse<Content> = {
  totalElements: number;
  totalPages: number;
  first: boolean;
  last: boolean;
  number: number;
  numberOfElements: number;
  size: number;
  empty: boolean;
  content: Content;
};
```

Hope it can be useful here

Reply:
Yeah, didn't see that. used